src/bsoncxx/array/view.cpp

```diff
@@ -139,6 +140,19 @@ element view::operator[](std::uint32_t i) const {
     return *(this->find(i));
 }
 
+view::const_iterator view::find_value(bsoncxx::types::bson_value::view v) const {
+    for (auto it = this->cbegin(); it != this->cend(); it++) {
+        if (it->get_value() == v) {
+            return it;
+        }
+    }
+    return this->cend();
+}
+
+bool view::has_value(bsoncxx::types::bson_value::view v) const {
```

Comment:
We could also overload `operator[]` to be consistent with the other find/operator methods for indices. So just:
```
element view::operator[](bsoncxx::types::bson_value::view v) const {
    return *(this->find_value(v));
}
```
would allow for boolean testing.

Reply:
I also don't think we should overload `operator[]`, I think we should try to keep methods in agreement with the `std::array` API as much as possible.  For `std::array`, `operator[]` functions as our existing one does, by looking up the element at the given index.

Reply:
That's a good, point. Resolved.